### PR TITLE
Rewrite MQTT tutorial

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tutorial_code/mqtt_mosquitto/mqtt_v1.toit [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2023 Toitware ApS.
+// Use of this source code is governed by a Zero-Clause BSD license that can
+// be found in the LICENSE_BSD0 file.
+
+import net
+import mqtt
+
+CLIENT_ID ::= ""
+HOST      ::= "test.mosquitto.org"
```
Comment (kasperl) [inline tutorial_code/mqtt_mosquitto/mqtt_v1.toit:9]: I'd probably avoid the alignment here.

File: docs/tutorials/mqtt/index.mdx [hunk 2]
```diff
@@ -1,30 +1,115 @@
...
+Insert the following code:
+<!-- ANALYZE CODE -->
 ```toit
 import net
 import mqtt
 
-// Include a random number, as we are sharing test.mosquitto.org with other users.
-CLIENT_ID ::= "my-client-id-$(random)"
+CLIENT_ID ::= ""  // Use a random client ID.
 HOST      ::= "test.mosquitto.org"
```
Comment (kasperl) [inline docs/tutorials/mqtt/index.mdx:113]: I'd probably avoid the alignment here.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: docs/tutorials/mqtt/index.mdx (hunk 2 above)
Comment (floitsch) [reply docs/tutorials/mqtt/index.mdx]: done.

File: tutorial_code/mqtt_mosquitto/mqtt_v1.toit (hunk 1 above)
Comment (floitsch) [reply tutorial_code/mqtt_mosquitto/mqtt_v1.toit]: done.

**Pull request merged**